lock-api/src/main/java/com/palantir/lock/client/LockRefreshingLockService.java

```diff
@@ -183,4 +207,26 @@ public void close() {
     public Set<LockRefreshToken> getAllRefreshingTokens() {
         return ImmutableSet.copyOf(toRefresh);
     }
+
+    @FunctionalInterface
+    public interface FailedRefreshCallback {
+        void onFailedRefresh(Set<LockRefreshToken> tokens);
+    }
+
+    /**
+     * Register a callback function to get called any time tokens fail to be refreshed.
+     *
+     * <p>Call {@link Disposable#dispose} on the returned value to remove the callback.
+     *
+     * @throws SafeIllegalArgumentException if the callback was already registered.
```

Comment:
Can we add a warning here that this API should not be used outside of large internal product unless you have the AtlasDB dev team's approval? I wouldn't want users in the metal product to end up using this.